src/App.tsx

```diff
@@ -80,17 +93,43 @@ const App = () => {
     [dispatch],
   )
 
-  useEffect(() => {
-    if (!isRestored && !isBackedUp) return
+  const loadInitialData = useCallback(async () => {
+    dispatch(restoreUser())
+    dispatch(fetchInitialData())
```

Comment:
I noticed this is calling to get the oracle price and current block height. Do we need that data before anything else in the app can load?

Its not too much, but in the old app we slowly got way too many things loading at the start and it was a really long launch process haha, so just want to be sure we dont just dump calls here.

Reply:
yeah, I hear what you're saying. @allenan is addressing data loading in a separate PR. Going to leave it as is for now.